src/basicio.cpp

```diff
@@ -236,15 +236,14 @@ namespace Exiv2
             return 0;
         }
 
+        if (fp_ == nullptr)
```

Comment:
This should be at the beginning of this function for consistency. Then the assert can go.

Reply:
Agreed. Can do.